Blog app: add API endpoints

brenoxav opened the pull request:
## Blog app: add API endpoints

- Create an API endpoint to list all posts.
- Create an API endpoint to list all comments for a post.
- API endpoints should receive JSON and respond JSON as well.

AshakaE commented on config/routes.rb:
```diff
@@ -10,7 +10,7 @@
-    resources :comments, only: [:create]
+    resources :comments, only: [:index, :create]
     resources :likes, only: [:create, :destroy]
   end
```
- [x] Kindly add endpoints for login and registration, I can only see index and show

StarSheriff2 replied:
We have added API authentication to the app. However, users can't sign up through the API, only sign in, so you will need to first create a user, before you are able to add a comment to a post, using the API.

vichuge replied:
- [x] As my companion said, is important to create the endpoint for register, I kindly invite you to add it to your project. 🐶 
![image](https://user-images.githubusercontent.com/20563252/136297213-193cbba1-c1c9-4908-9727-6d63240c0505.png)

vichuge replied:
Just do this and you'll finish this project. 🐱

StarSheriff2 replied:
Hi, we appreciate the feedback. Although this requirement was not clearly stated in the project description, we have implemented this feature. Users can now register through the Api.

AshakaE reviewed (changes requested):
### Hi Team 🙋‍♀️, 

Good job so far! 👌

Pull request is 👌 
Git flow ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


#  Required Changes ♻️ 🚧
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

StarSheriff2 commented:
Detailed instructions about how to use the API calls can be found in the README file. 😊

brenoxav commented:
Like Arturo said in the previous comment: 

> Although this requirement was not clearly stated in the project description, we have implemented this feature. Users can now register through the Api.

We added detailed instructions on how to use the API in the README.
